zcash_client_sqlite/src/lib.rs

```diff
@@ -611,35 +622,26 @@ impl<'a, P: consensus::Parameters> WalletWrite for DataConnStmtCache<'a, P> {
             }
 
             for output in &sent_tx.outputs {
-                match output.recipient_address {
-                    // TODO: Store the entire UA, not just the Sapling component.
-                    // This will require more info about the output index.
-                    RecipientAddress::Unified(ua) => wallet::insert_sent_note(
-                        up,
-                        tx_ref,
-                        output.output_index,
-                        sent_tx.account,
-                        ua.sapling().expect("TODO: Add Orchard support"),
-                        output.value,
-                        output.memo.as_ref(),
-                    )?,
-                    RecipientAddress::Shielded(addr) => wallet::insert_sent_note(
+                match &output.recipient {
+                    Recipient::Address(RecipientAddress::Transparent(addr)) => {
+                        wallet::insert_sent_utxo(
+                            up,
+                            tx_ref,
+                            output.output_index,
+                            sent_tx.account,
+                            addr,
+                            output.value,
+                        )?
+                    }
+                    shielded_recipient => wallet::insert_sent_note(
```

Comment:
This incorrectly resolves the above TODO and introduces a bug. `wallet::insert_sent_note` internally hard-codes Sapling as the pool, which made sense because it explicitly took in a `sapling::PaymentAddress`. This change coerces all shielded output types to Sapling, which will cause corruption and (in the case of mixed transactions) data loss.

The correct approach is provide what the TODO says: more information about the output index. We need to pass through the output pool type all thr way through from where `sent_tx.outputs` is populated. Or alternatively, map the UA to its Sapling case, preserving the approach we were taking before (deferring Orchard support to a later PR).

Reply:
I don't think this is correct. We need to be tracking the entire unified address to which we were requested to send the transaction; otherwise we're throwing away that critical information and can't retrieve it in the wallet history. This change was intentional. Right now, of course, we can only send to the Sapling pool, so that's why I didn't propagate the pool information through, but we need to retain the full UA somewhere.

Reply:
Fixed in https://github.com/zcash/librustzcash/pull/659/commits/179e22b76d8a34eff7cfc995683dcf34692816ac by propagating the pool information in from where it's definitively known.